kernel/comps/virtio/src/transport/pci/driver.rs

```diff
@@ -45,9 +49,22 @@ impl PciDriver for VirtioPciDriver {
         if device.device_id().vendor_id != VIRTIO_DEVICE_VENDOR_ID {
             return Err((BusProbeError::DeviceNotMatch, device));
         }
-        let transport = VirtioPciTransport::new(device)?;
-        let device = transport.pci_device().clone();
+
+        let device_id = *device.device_id();
+        let transport: Box<dyn VirtioTransport> = match device_id.device_id {
+            0x1000..0x1040 if (device.device_id().revision_id == 0) => {
+                // Transitional PCI Device ID in the range 0x1000 to 0x103f.
+                let legacy = VirtioPciLegacyTransport::new(device)?;
+                Box::new(legacy)
+            }
+            0x1040..0x107f => {
+                let modern = VirtioPciModernTransport::new(device)?;
```

Comment:
Since we only supply 0x1040 to 0x107f to the `VirtioPciModernTransport::new()`, I think we can refactor the device matching procedure in `VirtioPciModernTransport::new()`, considering the address range 0x1000 to 0x1040 as illegal input.